cpp/src/io/parquet/writer_impl.hpp

```diff
@@ -77,9 +77,15 @@ class writer::impl {
    *
    * @param table The set of columns
    * @param metadata The metadata associated with the table
+   * @param return_filemetadata If true, return the raw parquet file metadata
+   * @param metadata_out_file_path Column chunks file path to be set in the raw output metadata
    * @param stream Stream to use for memory allocation and kernels
+   * @return unique_ptr to FileMetadata thrift message if requested
    **/
-  void write(table_view const& table, const table_metadata *metadata, cudaStream_t stream);
+  std::unique_ptr<std::vector<uint8_t>> write(table_view const& table,
+                 const table_metadata *metadata,
+                 bool return_filemetadata, const std::string& metadata_out_file_path,
```

Comment:
@OlivierNV -  I appologize for not confirming this earlier...  The string passed in as `metadata_out_file_path` will **not** be included in the footer metadata written to the file - Only the output metadata will have the field populated, correct?

The naming convention you chose tells me you already know this:  It is not necessarily "correct" to set the "file_path" field in the footer metadata.  The reason for this is that the `"file_path"` value should correspond to the *relative* path of the data with respect to the metadata.

Reply:
Correct. The 'file_path' in the footer metadata remains empty. The value 'metadata_out_file_path' is only used in the metadata that is returned (it is inserted as is, so it is up to the caller to make it relative to the path of the data file)

Reply:
Perfect - Thanks for clarifying :)